packages/@ama-sdk/schematics/schematics/code-generator/open-api-cli-generator/open-api-cli.generator.ts

```diff
@@ -0,0 +1,85 @@
+import {CodeGenerator} from '../code-generator';
+import {spawn, SpawnOptions} from 'node:child_process';
+import {defaultTypescriptGeneratorOptions, OpenApiCliOptions} from './open-api-cli.options';
+
+/**
+ * Manage the schematic to generate a sdk using the @openapitools/openapi-generator-cli
+ */
+export class OpenApiCliGenerator extends CodeGenerator<OpenApiCliOptions> {
+  private packageManager = process.env && process.env.npm_execpath && process.env.npm_execpath.indexOf('yarn') === -1 ? 'npm' : 'yarn';
+  /** @inheritDoc */
+  protected generatorName = 'openapi-codegen-npm-cli';
+
+  /**
+   * Install the specified java open api generator
+   *
+   * @param version of the OpenApi Generator jar to use
+   * @param spawnOptions to configure your command line environment
+   */
+  private runInstallOpenApiGenerator(version: string, spawnOptions: SpawnOptions): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+      spawn(this.packageManager === 'npm' ? 'npx' : this.packageManager,
+        ['run', 'openapi-generator-cli', 'version-manager', 'set', version],
+        spawnOptions
+      ).on('close', (code: number) => {
+        if (code === 0) {
+          resolve();
+        } else {
+          reject(new Error(`OpenApiGeneratorCli failed to install OpenApiGenerator@${version}`));
+        }
+      });
+    });
+  }
+
+  /**
+   * Run the OpenApi generator jar with the configuration specified to generate a SDK based on the specified specification
+   *
+   * @param generatorOptions contains the version of the generator to use, the output dir, the specification file and config to use etc.
+   * @param spawnOptions to configure your command line environment
+   */
+  private runGenerator(generatorOptions: OpenApiCliOptions, spawnOptions: SpawnOptions) {
+    const args = [
+      'run',
+      'openapi-generator-cli',
+      'generate',
+      generatorOptions.generatorCustomPath ? `--custom-generator=${generatorOptions.generatorCustomPath}` : '',
+      '-g', generatorOptions.generatorName,
+      '-i', generatorOptions.specPath,
+      ...generatorOptions.specConfigPath ? ['-c', generatorOptions.specConfigPath] : [],
+      '-o', generatorOptions.outputPath
+    ];
+    return new Promise<void>((resolve, reject) => {
+      spawn(this.packageManager === 'npm' ? 'npx' : this.packageManager, args, spawnOptions)
+        .on('close', (code: number) => {
+          if (code === 0) {
+            resolve();
+          } else {
+            reject(new Error(`OpenApiGeneratorCli failed to run OpenApiGenerator with command 'run ${args.join(' ')}'`));
```

Comment:
```suggestion
            reject(new Error(`OpenApiGeneratorCli failed to run OpenApiGenerator with command '${args.join(' ')}'`));
```
as `run` is the 1st param of `args`